pkg/finder/finder.go

```diff
@@ -0,0 +1,87 @@
+package finder
+
+import (
+	"fmt"
+	"io/fs"
+	"log/slog"
+	"os"
+	"path/filepath"
+	"slices"
+
+	"github.com/getsops/sops/v3"
+	"github.com/getsops/sops/v3/stores/dotenv"
+	"github.com/getsops/sops/v3/stores/ini"
+	"github.com/getsops/sops/v3/stores/json"
+	"github.com/getsops/sops/v3/stores/yaml"
+)
+
+var supportedExtensions = []string{".ini", ".env", ".yaml", ".yml", ".json"}
+
+func getStore(fileType string) (sops.Store, error) {
+	switch fileType {
+	case ".ini":
+		return &ini.Store{}, nil
+	case ".env":
+		return &dotenv.Store{}, nil
+	case ".yaml", ".yml":
+		return &yaml.Store{}, nil
+	case ".json":
+		return &json.Store{}, nil
+	default:
+		return nil, fmt.Errorf("Unsupported file type: %s", fileType)
+	}
+}
+
+// SopsFiles searches a directory for YAML files and checks if they are valid SOPS files.
+// It returns a map of filepath with its sops.Metadata as value and an error.
+func SopsFiles(root string) (map[string]sops.Metadata, error) {
+	fileKeyMap := make(map[string]sops.Metadata)
+
+	err := filepath.WalkDir(root, func(path string, d fs.DirEntry, err error) error {
+		if err != nil {
+			return err
+		}
+
+		// Skip directories
+		if d.IsDir() {
+			return nil
+		}
+
+		ext := filepath.Ext(d.Name())
+
+		if slices.Contains(supportedExtensions, ext) {
+			encrypteddata, err := os.ReadFile(path)
+			if err != nil {
+				slog.Warn("Could not read file:", "filepath", path, "error", err)
+				return nil
+			}
+
+			store, err := getStore(ext)
+			if err != nil {
+				return nil
+			}
```

Comment:
You could flip the order here to first call `getStore` (light operation) and only afterwards call `os.ReadFile` (heavy operation) if there's a store for the extension.

That way you could get rid of the `supportedExtensions` slice (which needs to be in sync with the cases in `getStore`) and then call to `slices.Contains(supportedExtensions, ext)` entirely, which simplifies the code a bit.